extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/MSQTestBase.java

```diff
@@ -1010,6 +1041,23 @@ public void verifyPlanningErrors()
       assertThat(e, expectedValidationErrorMatcher);
     }
 
+    protected void verifyLookupLoadingInfoInTaskContext(Map<String, Object> context)
+    {
+      String lookupLoadingMode = context.get(PlannerContext.CTX_LOOKUP_LOADING_MODE).toString();
+      List<String> lookupsToLoad = (List<String>) context.get(PlannerContext.CTX_LOOKUPS_TO_LOAD);
+      if (expectedLookupLoadingSpec != null) {
+        Assert.assertEquals(expectedLookupLoadingSpec.getMode().toString(), lookupLoadingMode);
+        if (expectedLookupLoadingSpec.getMode().equals(LookupLoadingSpec.Mode.ONLY_REQUIRED)) {
+          Assert.assertEquals(new ArrayList<>(expectedLookupLoadingSpec.getLookupsToLoad()), lookupsToLoad);
+        } else {
+          Assert.assertNull(lookupsToLoad);
+        }
+      } else {
+        Assert.assertEquals(LookupLoadingSpec.Mode.NONE.toString(), lookupLoadingMode);
+        Assert.assertNull(lookupsToLoad);
+      }
+    }
```

Comment:
Rather than this if-else chain, we can just build a `LookupLoadingSpec` object from the values in the context and then do an equals check with the expected lookup loading spec. You would also need to override equals and hashCode in LookupLoadingSpec.

Reply:
I can try taking care of this in the next PR with compaction changes, hope that works.

Reply:
Absolutely, thanks!